### DISCUSSION TITLE
DeprecationWarnings and FutureWarnings when developing on a Windows Server

### DISCUSSION THREAD

**Discussion by jelleooms** [apache/airflow#27340] [General]
Warnings when import anything from airflow:
`from airflow.models.dagrun import DagRun`

`
C:\Users\USER\Source\Repos\edm\venv_new\lib\site-packages\airflow\configuration.py:549: DeprecationWarning: The sql_alchemy_conn option in [core] has been moved to the sql_alchemy_conn option in [database] - the old setting has been used, but please update your config.
  option = self._get_option_from_config_file(deprecated_key, deprecated_section, key, kwargs, section)
C:\Users\USER\Source\Repos\edm\venv_new\lib\site-packages\airflow\configuration.py:443: FutureWarning: The 'dag_default_view' setting in [webserver] has the old default value of 'tree'. This value has been changed to 'grid' in the running config, but please update your config before Apache Airflow 3.0.
  warnings.warn(
C:\Users\USER\Source\Repos\edm\venv_new\lib\site-packages\airflow\configuration.py:443: FutureWarning: The 'log_filename_template' setting in [logging] has the old default value of '{{ ti.dag_id }}/{{ ti.task_id }}/{{ ts }}/{{ try_number }}.log'. This value has been changed to 'dag_id={{ ti.dag_id }}/run_id={{ ti.run_id }}/task_id={{ ti.task_id }}/{% if ti.map_index >= 0 %}map_index={{ ti.map_index }}/{% endif %}attempt={{ try_number }}.log' in the running config, but please update your config before Apache Airflow 3.0.
  warnings.warn(
C:\Users\USER\Source\Repos\edm\venv_new\lib\site-packages\airflow\configuration.py:549: DeprecationWarning: The auth_backend option in [api] has been renamed to auth_backends - the old setting has been used, but please update your config.
  option = self._get_option_from_config_file(deprecated_key, deprecated_section, key, kwargs, section)
C:\Users\USER\Source\Repos\edm\venv_new\lib\site-packages\airflow\configuration.py:443: FutureWarning: The 'auth_backends' setting in [api] has the old default value of 'airflow.api.auth.backend.deny_all'. This value has been changed to 'airflow.api.auth.backend.session' in the running config, but please update your config before Apache Airflow 3.0.
  warnings.warn(
C:\Users\USER\Source\Repos\edm\venv_new\lib\site-packages\airflow\configuration.py:443: FutureWarning: The 'log_id_template' setting in [elasticsearch] has the old default value of '{dag_id}-{task_id}-{execution_date}-{try_number}'. This value has been changed to '{dag_id}-{task_id}-{run_id}-{map_index}-{try_number}' in the running config, but please update your config before Apache Airflow 3.0.
  warnings.warn(
WARNING:root:OSError while attempting to symlink the latest log directory
C:\Users\USER\Source\Repos\edm\venv_new\lib\site-packages\airflow\configuration.py:549 DeprecationWarning: The sql_alchemy_pool_enabled option in [core] has been moved to the sql_alchemy_pool_enabled option in [database] - the old setting has been used, but please update your config.
C:\Users\USER\Source\Repos\edm\venv_new\lib\site-packages\airflow\configuration.py:549 DeprecationWarning: The sql_engine_encoding option in [core] has been moved to the sql_engine_encoding option in [database] - the old setting has been used, but please update your config.
C:\Users\USER\Source\Repos\edm\venv_new\lib\site-packages\airflow\configuration.py:549 DeprecationWarning: The sql_alchemy_schema option in [core] has been moved to the sql_alchemy_schema option in [database] - the old setting has been used, but please update your config.
C:\Users\USER\Source\Repos\edm\venv_new\lib\site-packages\airflow\configuration.py:549 DeprecationWarning: The max_db_retries option in [core] has been moved to the max_db_retries option in [database] - the old setting has been used, but please update your config.DASFFdsF
`

Installed packages:
adal==1.2.7
alembic==1.7.7
anyio==3.5.0
apache-airflow==2.4.2
apache-airflow-providers-common-sql==1.2.0
apache-airflow-providers-ftp==3.1.0
apache-airflow-providers-http==4.0.0
apache-airflow-providers-imap==3.0.0
apache-airflow-providers-sqlite==3.2.1
apispec==3.3.2
argcomplete==1.12.3
argon2-cffi==21.3.0
argon2-cffi-bindings==21.2.0
attrs==22.1.0
azure-common==1.1.28
azure-core==1.22.1
azure-data-tables==12.2.0
azure-graphrbac==0.61.1
azure-identity==1.7.1
azure-mgmt-core==1.3.0
azure-mgmt-sql==3.0.1
azure-storage-common==2.1.0
azure-storage-file==2.1.0
Babel==2.10.1
bcpandas==2.0.0
bcrypt==3.2.2
beautifulsoup4==4.11.1
blinker==1.4
bs4==0.0.1
cached-property==1.5.2
cachelib==0.9.0
cattrs==22.2.0
certifi==2022.5.18.1
cffi==1.15.0
chardet==3.0.4
charset-normalizer==2.0.12
click==8.1.3
clickclick==20.10.2
colorama==0.4.4
colorlog==4.0.2
commonmark==0.9.1
configparser==3.5.3
ConfigUpdater==3.1.1
connexion==2.13.1
construct==2.10.68
cron-descriptor==1.2.24
croniter==0.3.37
cryptography==37.0.2
defusedxml==0.7.1
Deprecated==1.2.13
dill==0.3.5.1
dnspython==2.2.0
docutils==0.16
email-validator==1.1.3
exceptiongroup==1.0.0rc9
Flask==2.2.2
Flask-Admin==1.6.0
Flask-AppBuilder==4.1.4
Flask-Babel==2.0.0
Flask-Caching==2.0.1
Flask-JWT-Extended==4.4.4
Flask-Login==0.6.2
Flask-OpenID==1.3.0
Flask-Session==0.4.0
Flask-SQLAlchemy==2.5.1
flask-swagger==0.2.14
Flask-WTF==1.0.1
funcsigs==1.0.2
future==0.18.2
graphviz==0.19.1
greenlet==1.1.2
gunicorn==20.1.0
h11==0.12.0
httpcore==0.14.7
httpx==0.22.0
idna==2.10
importlib-metadata==4.11.1
importlib-resources==1.5.0
inflection==0.5.1
iso8601==1.0.2
isodate==0.6.1
itsdangerous==2.1.2
Jinja2==3.1.2
json-merge-patch==0.2
jsonschema==3.2.0
lazy-object-proxy==1.4.3
linkify-it-py==2.0.0
lockfile==0.12.2
lxml==4.8.0
Mako==1.2.0
Markdown==3.3.7
markdown-it-py==2.1.0
MarkupSafe==2.1.1
marshmallow==3.15.0
marshmallow-enum==1.5.1
marshmallow-oneofschema==3.0.1
marshmallow-sqlalchemy==0.26.1
mdit-py-plugins==0.3.1
mdurl==0.1.2
msal==1.20.0
msal-extensions==0.3.1
msrest==0.6.21
msrestazure==0.6.4
natsort==8.1.0
ntlm-auth==1.5.0
numpy==1.22.4
oauthlib==3.2.0
openapi-schema-validator==0.2.3
openapi-spec-validator==0.4.0
operators==1.0.1
packaging==21.3
pandas==1.5.1
paramiko==2.11.0
pathspec==0.9.0
pendulum==2.1.2
pluggy==1.0.0
portalocker==2.4.0
prison==0.2.1
psutil==5.9.1
psycopg2==2.9.3
pycparser==2.21
pycryptodomex==3.14.1
Pygments==2.12.0
PyJWT==2.6.0
pykeepass==4.0.2
PyNaCl==1.5.0
pyodbc==4.0.34
pyparsing==3.0.9
pyrsistent==0.18.1
pysftp==0.2.9
python-daemon==2.3.0
python-dateutil==2.8.2
python-nvd3==0.15.0
python-slugify==6.1.2
python3-openid==3.2.0
pytz==2022.1
pytzdata==2020.1
pywin32==304
PyYAML==6.0
requests==2.27.1
requests-ntlm==1.1.0
requests-oauthlib==1.3.1
requests-toolbelt==0.10.1
rfc3986==1.5.0
rich==12.4.4
setproctitle==1.2.3
six==1.16.0
sniffio==1.2.0
soupsieve==2.3.2.post1
SQLAlchemy==1.4.42
SQLAlchemy-JSONField==1.0.0
SQLAlchemy-Utils==0.38.3
sqlparse==0.4.3
sshtunnel==0.4.0
swagger-ui-bundle==0.0.9
tabulate==0.8.9
tenacity==8.0.1
termcolor==1.1.0
text-unidecode==1.3
thrift==0.16.0
typing-extensions==4.2.0
tzlocal==1.5.1
uc-micro-py==1.0.1
unicodecsv==0.14.1
urllib3==1.25.11
Werkzeug==2.2.2
wrapt==1.14.1
WTForms==2.3.3
zipp==3.7.0
zope.deprecation==4.4.0

Airflow is not running on this Windows Server nor is it my intention to do so. I would like to know how I can get rid of these warnings. 
I have installed airflow on ubuntu servers with the correct config file (airflow.cfg), but adding that file to my local code in Windows doesn't do anything so far.

**Comment by potiuk** [comment]
The warnigs seem pretty self-descriptive and they even tell you what to do.

You have to correct thesettings - your settings are simply old (likely in arflow.cfg). Your best bet is to find where they are defined, remove them and reset-them from scratch (you are likely using old config updated and maintained from older Airflow.

Note that there might be other, unfreseen problems on Windows as (as of yet) Airflow is not supported on Windows natively, so your expectations about getting help here might not be met- mostly because there is pretty much no-one who runs in on Windows, so if you will have any other problems, there are no people (neither maintainers nor users) who will be willing to spend their time to try to reproduce it.

Generally speaking - we have an open issue https://github.com/apache/airflow/issues/10388 to make Airlfow on Windows - but the state here is that we are rather expecting people like you to investigate problems they will have on Windows and dig into the code, find the solutions and provide PRs, rather than the other way round - where they might expect some help when they run into troubles.

So if you are not ready to go in this direction and rather investigate things on your own, I strongly recommend you use WSL2 and run Airlfow in Linux run on top of Windows. This is much more stable and there are lots of people using it who might help you.
